packages/connection-info/src/connection-info.ts

```diff
@@ -12,24 +12,31 @@ export interface ConnectionInfo {
   lastUsed?: Date;
 
   /**
-   * If present the connection is marked as a favorite by the user.
+   * True if the connection is marked as favorite.
    */
-  favorite?: ConnectionFavoriteOptions;
+  userFavorite?: boolean;
 
   /**
-   * The options used to connect
+   * Name of the connection.
    */
-  connectionOptions: ConnectionOptions;
-}
+  name?: string;
 
-export interface ConnectionFavoriteOptions {
   /**
-   * User-defined name of the connection.
+   * Colour of the connection, if any;
    */
-  name: string;
+  color?: string;
 
   /**
-   * Hex-code of the user-defined color.
+   * Legacy favourite information
    */
-  color?: string;
+  readonly favorite?: ConnectionFavoriteOptions;
+  /**
+   * The options used to connect
+   */
+  connectionOptions: ConnectionOptions;
+}
+
+interface ConnectionFavoriteOptions {
+  readonly name: string;
+  readonly color?: string;
 }
```

Comment:
What does it mean to be a named, but not a favorite connection? "Favorite" so far was always just our word for "Saved", we just didn't rename it to that because there was no real value other than maybe some naming clarity in it.

If we're introducing a new type of connection here, would it make sense to keep the `favorite` field as it was before, just add a boolean flag that indicates whether this is a connection of the third kind (saved, unnamed, named-but-not-saved)?

Reply:
With the multiple connections project, you can specify the name of the connection and the colour without marking it as favourite. Now we are saving all connections (recent + favourites) and they can have their own name/color.

The behaviour is a bit different, that is why we need to change a bit the format.

Reply:
Would there be an obvious downside to leaving the format as-is as much as possible? That would mean that older Compass versions will still be able to understand new files, just wrongly assume that some connections are favorites when they weren't supposed to be, which seems to be fine to me

Reply:
With the format as we have right now, the only way to specify the name or colour is by specifying the ConnectionFavouriteOptions, and our logic assumes that if the field favourite exists, it's a favourite connection. So essentially, any connection would be favourite by default or we would need to add a new field.

Right now what I'm doing here is to adapt the old favourites to the new format when loading them, so if you change them and save them again, it will use the new format, so we `lazily migrate connections on demand`.

It's in connection-storage.ts#235.

Reply:
> or we would need to add a new field

Yes, that was what I was suggesting :)

> Right now what I'm doing here is to adapt the old favourites to the new format when loading them, so if you change them and save them again, it will use the new format, so we `lazily migrate connections on demand`.

Yeah, I did see that. I'm just saying, this looks like it would work without the need for a migration (and in particular a unidirectional one that Compass users who downgrade couldn't benefit from) fairly easily.

Reply:
So you suggest to use a new field, maybe userFavorite or something like that? I think it would be a bit misleading with the namings, as we already have isFavourite, favourite and now userFavourite and two of them would be used together.

Reply:
Right, we'd want to be cautious about the naming. (Where is `isFavourite`? That's not a property of the format itself, is it? `git grep` didn't show anything)

I'd maybe use a `savedConnectionType: 'favorite' | 'named'` or something like that.

And I do understand that the naming isn't optimal, it has not been ever since we moved away from the `favorite` terminology in the UI, but this is a storage format and that's sometimes just a part of storage formats evolving over time

Reply:
It's in connection-storage.ts#48:

```js
const ConnectionSchema: z.Schema<ConnectionWithLegacyProps> = z
  .object({
    _id: z.string().uuid().optional(),
    isFavorite: z.boolean().optional(),
    name: z.string().optional(),
    connectionInfo: z
```

I think it's from an old migrated storage. I think we can work with savedConnectionType, for some reason I didn't think about having an string literal type with the type of saved connection. That sounds like a great idea.

Will do the change.